src/fn/watch-project.js

```diff
@@ -0,0 +1,160 @@
+            await uploadResources(projectPath);
+            callback(fileName, true);
+        } else if (fileName.match(/messages-[\w]*\.properties/)) {
+            await uploadCustomTranslations(environment.apiUrl, projectPath, environment.skipTranslationCheck);
+            callback(fileName, true);
+        } else if (fileName.match(/.*\.js$/) || fileName.match(/.*\.json$/)) {
+            await compileAppSettings(projectPath);
+            await uploadAppSettings(api, projectPath);
+            callback(fileName, true);
+        } else {
+            warn('main listen: don\'t know what to do with', fileName);
+        }
+    };
+};
+
+let appFormWait = false;
+const appFormListener = function (projectPath, callback) {
+    return async (event, fileName) => {
+        if (event !== 'change' || appFormWait) {
+            return;
+        }
+        appFormWait = setTimeout(() => { appFormWait = false; }, DEBOUNCE_DELAY);
+        if (fileName.match(formXLSRegex) || fileName.match(formPropertiesRegex)) {
+            await convertForms(projectPath, 'app', {
+                enketo: true,
+                forms: [fileName.split('.')[0]],
+                transformer: xml => xml.replace('</instance>', '</instance>\n      <instance id="contact-summary"/>'),
+            });
+            await uploadForms(projectPath, 'app', {
+                forms: [fileName.split('.')[0]],
+            });
+            callback(fileName, true);
+        } else if (fileName.match(formMediaRegex)) {
+            const absDirPath = path.join(projectPath, 'forms', 'app', fileName);
+            watchFormMediaDir(fileName, absDirPath, projectPath);
+        } else {
+            warn('don\'t know what to do with', fileName);
+        }
+    };
+};
+
+const formMediaListener = function (form, projectPath) {
+    return async () => {
+        await uploadForms(projectPath, 'app', {
+            forms: [form],
+        });
+    };
+};
+
+function watchFormMediaDir(dirName, absDirPath, projectPath) {
+    if (!fs.existsSync(absDirPath) || !fs.lstatSync(absDirPath).isDirectory()) return;
+    return fs.watch(absDirPath, formMediaListener(dirName.split('-')[0], projectPath));
+}
+
+let contactFormWait = false;
+const contactFormListener = function (projectPath, callback) {
+    return async (event, fileName) => {
+        if (event !== 'change' || contactFormWait) {
+            return;
+        }
+        contactFormWait = setTimeout(() => { contactFormWait = false; }, DEBOUNCE_DELAY);
+        if (fileName.match(formXLSRegex)) {
+            await convertContactForm(projectPath, [fileName.split('.')[0]]);
+            await uploadForms(projectPath, 'contact', {
+                id_prefix: 'contact:',
+                forms: [fileName.split('.')[0]],
+                default_context: { person: false, place: false },
+            });
+            callback(fileName, true);
+        } else {
+            if (!fileName.match(formXMLRegex)) {
```

Comment:
How come the XML here is treated differently than app forms?
Again, I changed the XML directly and nothing happened (this time I didn't even get an error).